npm_and_yarn/spec/dependabot/npm_and_yarn/update_checker_spec.rb

```diff
@@ -1491,6 +1491,180 @@
+
+        expect(described_class::VulnerabilityAuditor).
+          to receive(:new).
+          with(
+            dependency_files: dependency_files,
+            credentials: credentials,
+            allow_removal: false
+          ).and_call_original
+
+        conflicting_dependencies_result = checker.send(:conflicting_dependencies)
+
+        expect(conflicting_dependencies_result.count).to eq(1)
+        expect(conflicting_dependencies_result.first).
+          to eq(
+            "explanation" => "@dependabot-fixtures/npm-parent-dependency@2.0.0 requires " \
+                             "@dependabot-fixtures/npm-transitive-dependency@1.0.0 via " \
+                             "@dependabot-fixtures/npm-intermediate-dependency@0.0.1",
+            "name" => "@dependabot-fixtures/npm-intermediate-dependency",
+            "requirement" => "1.0.0",
+            "version" => "0.0.1"
+          )
+      end
+    end
+
+    context "with a conflicting dependency and an unsatisfiable vulnerablity" do
+      let(:dependency_version) { "1.0.0" }
+      let(:target_version) { Dependabot::NpmAndYarn::Version.new("1.0.1") }
+      let(:security_advisories) do
+        [
+          Dependabot::SecurityAdvisory.new(
+            dependency_name: "@dependabot-fixtures/npm-transitive-dependency",
+            package_manager: "npm_and_yarn",
+            vulnerable_versions: ["< 1.0.3"]
+          )
+        ]
+      end
+
+      it "delegates to the ConflictingDependencyResolver and VulnerabilityAuditor and explains the conflict", :vcr do
+        expect(described_class::ConflictingDependencyResolver).
+          to receive(:new).
+          with(
+            dependency_files: dependency_files,
+            credentials: credentials
+          ).and_call_original
+
+        expect(described_class::VulnerabilityAuditor).
+          to receive(:new).
+          with(
+            dependency_files: dependency_files,
+            credentials: credentials,
+            allow_removal: false
+          ).and_call_original
+
+        conflicting_dependencies_result = checker.send(:conflicting_dependencies)
+
+        expect(conflicting_dependencies_result.count).to eq(2)
+
+        expect(conflicting_dependencies_result.first).
+          to eq(
+            "explanation" => "@dependabot-fixtures/npm-parent-dependency@2.0.0 requires " \
+                             "@dependabot-fixtures/npm-transitive-dependency@1.0.0 via " \
+                             "@dependabot-fixtures/npm-intermediate-dependency@0.0.1",
+            "name" => "@dependabot-fixtures/npm-intermediate-dependency",
+            "requirement" => "1.0.0",
+            "version" => "0.0.1"
+          )
+
+        expect(conflicting_dependencies_result.last).
+          to eq(
+            "dependency_name" => "@dependabot-fixtures/npm-transitive-dependency",
+            "explanation" => "No patched version available for @dependabot-fixtures/npm-transitive-dependency",
```

Comment:
What would the explanation be if it's one of the parents that doesn't have an update to a non locking version? I think that'd be the primary use case of this error as we might not get here if the vulnerable dependency doesn't have a fix available.

Reply:
I think that's captured by this test in the vulnerability auditor https://github.com/dependabot/dependabot-core/pull/5645/files#diff-f0c4f0847c642ca1ba8627d73ce6dfc76f107816ac9916ef12d3ae5c4edce665R122-R149

The error message is the same though. Would you expect the message to list out the locking parent?

Reply:
I removed the test for "with an unsatisfiable vulnerability" because I don't think we can reach that code without also having conflicting dependencies.

I updated the test with both "conflicting dependencies and an unsatisfiable vulnerability" to use an invalid version range for the vulnerability in https://github.com/dependabot/dependabot-core/pull/5645/commits/a8d6b3a1f28a5ea5c1759e01ad22737bfe73024c. 

Our @dependabot-fixtures setup is not complex enough to test this scenario using valid vulnerability ranges, but it does seem like a case that can occur in a project, so I want to keep test coverage around it.

Reply:
I opened https://github.com/dependabot/dependabot-core/pull/5672 to create and test a scenario where a valid vulnerable version range is blocked by a locking parent dependency